### PR TITLE
Make the logger configurable everywhere

### LINKED ISSUE #82: Make the swarrot logger configurable
Swarrot logs a lot. It can be useful to be able to use a custom logger (with custom logging rules) to be able to configure the verbosity.

**TODO:**
* Deprecate the `processor_logger` configuration key
* Create a new `logger` configuration to replace the previous one
* Use it everywhere (even in the command which use the default logger for now)

### PR DESCRIPTION
**Pull request by odolbeau**
Fix #82

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -46,7 +57,16 @@ public function getConfigTreeBuilder()
                 ->end()
                 ->scalarNode('default_connection')->defaultValue(null)->end()
                 ->scalarNode('default_command')->defaultValue('swarrot.command.base')->cannotBeEmpty()->end()
-                ->scalarNode('publisher_logger')->defaultValue('swarrot.logger.null')->cannotBeEmpty()->end()
+                ->scalarNode('publisher_logger')
+                    ->validate()
+                    ->always()
+                        ->then(function ($v) use(&$defaultLogger) {
+                            $defaultLogger = $v;
```
Comment (sroze) [inline DependencyInjection/Configuration.php]: Why this `$defaultLogger` variable?

Comment (odolbeau) [inline DependencyInjection/Configuration.php]: A failed attempt! Thanks! :)

**Comment by lyrixx** [comment]
You forgot an update in the changelog / upgrade file

**Comment by lyrixx** [comment]
:+1:

**Comment by sroze** [comment]
You don't have any test on the `logger` node, but only on your _legacy_ `publisher_logger` node. I would add one.

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -35,6 +35,17 @@ public function getConfigTreeBuilder()
         $knownProcessors = $this->knownProcessors;
 
         $rootNode
+            ->beforeNormalization()
+                ->always()
+                ->then(function ($v) {
+                    // Deal with old logger config
+                    if (isset($v['publisher_logger']) && !isset($v['logger'])) {
+                        $v['logger'] = $v['publisher_logger'];
```
Comment (stof) [inline DependencyInjection/Configuration.php]: you should also trigger a deprecation notice here, to let developers know they should update their config

Comment (stof) [inline DependencyInjection/Configuration.php]: ah, I saw that you triggered it on ``validate``

File: Tests/DependencyInjection/SwarrotExtensionTest.php [hunk 3]
```diff
@@ -90,6 +90,18 @@ public function test_it_registers_the_collector_when_explicitly_enabled()
         $this->assertHasService($container, 'swarrot.data_collector');
     }
 
+    public function test_it_use_the_asked_logger()
+    {
+        $container = $this->createContainer(false);
+
+        $this->loadConfig($container, array('publisher_logger' => 'my_awesome_logger'));
```
Comment (stof) [inline Tests/DependencyInjection/SwarrotExtensionTest.php]: As this test uses the deprecated config, you should mark it as ``@group legacy`` to be compatible with the usage of the PHPUnit bridge to handle usage of deprecated APIs in tests (and you should add the bridge as a dev requirement if it is not there yet IMO)

**Comment by odolbeau** [comment]
Tests updated. :)

**Comment by sroze** [comment]
It looks like `symfony/phpunit-bridge` drop the PHP 5.3 support so the tests are broken @odolbeau... :/

**Comment by odolbeau** [comment]
Fixed now @sroze!

**Comment by sroze** [comment]
Looks good then :+1:

**Pull request merged**